src/providers/gdal/qgsgdalprovider.cpp

```diff
@@ -1082,11 +1085,52 @@ QgsRasterIdentifyResult QgsGdalProvider::identify( const QgsPointXY &point, QgsR
       else
         results.insert( i, value );
     }
-    delete myBlock;
   }
   return QgsRasterIdentifyResult( QgsRaster::IdentifyFormatValue, results );
 }
 
+bool QgsGdalProvider::worldToPixel( double x, double y, int &col, int &row ) const
+{
+  double div = ( mGeoTransform[2] * mGeoTransform[4] - mGeoTransform[1] * mGeoTransform[5] );
+  if ( div < 2 * std::numeric_limits<double>::epsilon() )
+    return false;
+  double doubleCol = -( mGeoTransform[2] * ( mGeoTransform[3] - y ) + mGeoTransform[5] * x - mGeoTransform[0] * mGeoTransform[5] ) / div;
+  double doubleRow = ( mGeoTransform[1] * ( mGeoTransform[3] - y ) + mGeoTransform[4] * x - mGeoTransform[0] * mGeoTransform[4] ) / div;
```

Comment:
A part of the equation could be simplified to ( x - mGeoTransform[0] ) * ( ... )